packages/bot-config-utils/src/bot-config-utils.ts

```diff
@@ -28,33 +28,113 @@ type Conclusion =
   | 'action_required'
   | undefined;
 
-export interface OctokitContext {
-  octokit: Octokit;
+/**
+ * This interface is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+interface File {
+  content: string | undefined;
+}
+
+/**
+ * This function is used for comforting typescript's type system to
+ * deal with the response from `octokit.repos.getContent`.
+ */
+function isFile(file: File | unknown): file is File {
+  return (file as File).content !== undefined;
 }
 
-export interface ValidateConfigResult {
+/**
+ * The return value of `validateConfig` function.
+ *
+ * @template T
+ * @prop {boolean} isValid - The result of the schema validation.
+ * @prop {string} errorText - The error message from the validation.
+ * @prop {T} config - The loaded config object,only available when the
+ *   validation succeeds.
+ */
+export interface ValidateConfigResult<T> {
```

Comment:
I'd find the code easier to read if `T` was renamed `ConfigType`.

Reply:
Well, but it's mostly the naming convention to use T for type template.